lib/pf/client/queue.pm

```diff
@@ -0,0 +1,117 @@
+package pf::client::queue;
+
+=head1 NAME
+
+pf::client::queue add documentation
+
+=cut
+
+=head1 DESCRIPTION
+
+pf::client::queue
+
+=cut
+
+use strict;
+use warnings;
+use Moo;
+use Redis::Fast;
+use pf::file_paths;
+use List::MoreUtils qw(all);
+use Sereal::Encoder qw(sereal_encode_with_object);
+use pf::Sereal qw($ENCODER);
+use pf::task;
+
+our $DEFAULT_EXPIRATION = 300;
+
+has redis => (
+    is      => 'rw',
+    builder => 1,
+    lazy    => 1,
+);
+
+has server => (
+    is       => 'rw',
+    required => 1,
+    default  => sub { "127.0.0.1:6380" },
+);
+
+sub _build_redis {
+    my ($self) = @_;
+    my %args;
+    my $server = $self->server;
+
+    #If there is a / then consider it unix socket
+    if ($server =~ m#/#) {
+        $args{sock} = $server;
+    }
+    else {
+        $args{server} = $server;
+    }
+    return Redis::Fast->new(%args);
+}
+
+sub submit {
+    my ($self, $queue, $task_type, $task_data, $expire_in) = @_;
+    $expire_in //= $DEFAULT_EXPIRATION;
+    my $id    = pf::task->generateId();
+    my $redis = $self->redis;
+    # Batch the creation of the task and it's ttl and placing it on the queue to improve performance
+    $redis->multi(sub {});
+    $redis->hmset($id, data => sereal_encode_with_object($ENCODER, [$task_type, $task_data]), expire => $expire_in, sub {});
+    $redis->expire($id, $expire_in, sub {});
+    $redis->lpush("Queue:$queue", $id, sub {});
+    $redis->hincrby("TaskCount", $task_type, 1, sub {});
```

Comment:
I think a per-queue counter will be more than useful because we have the use case of having multiple queues right now with the pfdhcplistener needing it's own queue